src/psyclone/gocean1p0.py

```diff
@@ -947,74 +948,122 @@ def __init__(self, call, parent=None):
+                        depth = arg.stencil.depth(i, j)
+                        for offset in range(1, depth+1):
+                            if i == 0 and j == 0:
+                                # In this case it just refers to pointwise
+                                offset = 0
+
+                            inner_index = Reference(inner_symbol)
+                            outer_index = Reference(outer_symbol)
+                            if j > 0:
+                                inner_index = BinaryOperation.create(
+                                    BinaryOperation.Operator.ADD,
+                                    inner_index,
+                                    Literal(str(abs(offset)), INTEGER_TYPE))
+                            elif j < 0:
+                                inner_index = BinaryOperation.create(
+                                    BinaryOperation.Operator.SUB,
+                                    inner_index,
+                                    Literal(str(abs(offset)), INTEGER_TYPE))
+
+                            if i > 0:
+                                outer_index = BinaryOperation.create(
+                                    BinaryOperation.Operator.ADD,
+                                    outer_index,
+                                    Literal(str(abs(offset)), INTEGER_TYPE))
+                            elif i < 0:
+                                outer_index = BinaryOperation.create(
+                                    BinaryOperation.Operator.SUB,
+                                    outer_index,
+                                    Literal(str(abs(offset)), INTEGER_TYPE))
+
+                            access = StructureReference.create(symbol, [
+                                ("data", [inner_index, outer_index])
+                            ])
+                            if arg.access in (AccessType.WRITE,
+                                              AccessType.READWRITE):
+                                # if write_access is not None:
+                                #     raise InternalError(
+                                #         "This is not a valid kernel, a "
+                                #         "kernel can only write to one field"
+                                #     )
+                                write_access = access.copy()
+                            if arg.access != AccessType.WRITE:
+                                read_accesses.append(access)
+
+        # Now create the assignment prototype
+        # if not write_access:
+        #     raise InternalError(
+        #         "This is not a valid kernel, a kernel must write "
+        #         "to one field.")
+        if not read_accesses:
+            # There can be write-only kernels, in this case we just put
+            # a constant literal
+            rhs = Literal("1", INTEGER_TYPE)
+        else:
+            rhs = read_accesses.pop(0)
+            while read_accesses:
```

Comment:
I have to admit I felt a bit uneasy about creating a proper Fortran expression (`write=read+read+...`), but I can't find anything wrong with that, and indeed it should have the beauty that the variable access sequences (and then the DA) just work out of the box.